src/credentials/FabricTable.cpp

```diff
@@ -317,12 +317,24 @@ CHIP_ERROR FabricInfo::VerifyCredentials(const ByteSpan & noc, const ByteSpan &
     NodeId nodeId;
     ReturnErrorOnFailure(ExtractNodeIdFabricIdFromOpCert(certificates.GetLastCert()[0], &nodeId, &fabricId));
 
+    FabricId icacFabricId = kUndefinedFabricId;
     if (!icac.empty())
     {
-        FabricId icacFabric = kUndefinedFabricId;
-        if (ExtractFabricIdFromCert(certificates.GetCertSet()[1], &icacFabric) == CHIP_NO_ERROR && icacFabric != kUndefinedFabricId)
+        if (ExtractFabricIdFromCert(certificates.GetCertSet()[1], &icacFabricId) == CHIP_NO_ERROR &&
```

Comment:
Is the indexing here always safe? I do not see a check that guarantees me that indexing directly like this will hit a proper cert in this case. This may be OK based on the API contract, but is there something we can do to make sure?

Reply:
In the context of this PR it is guaranteed that indexing reach the right certificate. However, in a more general case I'll think about proper updates and may address it in a separate PR.